File: game-server/src/main/java/org/forfun/mmorpg/game/id/IdKey.java
package org.forfun.mmorpg.game.id;

public enum IdKey {

    /**
     * 玩家
     */
    PLAYER(1),

    /**
     * 道具
     */
    ITEM(2),

    /**
     * 战场房间号
     */
    MATCH(3, false),

    ;

    int id;

    /**
     * 是否需要持久化
     */
    boolean saveToDb;

    IdKey(int id) {
        this.id = id;
    }

    IdKey(int id, boolean saveToDb) {
        this.id = id;
        this.saveToDb = saveToDb;
    }

}

File: game-server/src/main/java/org/forfun/mmorpg/game/id/IdKey.kt
package org.forfun.mmorpg.game.id

enum class IdKey(
    val id: Int,
    /**
     * 是否需要持久化
     */
    val saveToDb: Boolean = true
) {
    /**
     * 玩家
     */
    PLAYER(1),

    /**
     * 道具
     */
    ITEM(2),

    /**
     * 战场房间号
     */
    MATCH(3, false)
}
